src/DocumentFormat.OpenXml/OpenXmlElement.cs

```diff
@@ -962,25 +962,32 @@ public IEnumerable<OpenXmlElement> Descendants()
 
             while (true)
             {
-                if (stack.Peek() == root && stack.Peek().FirstChild != null)
-                {
-                    root = stack.Peek().FirstChild;
-                    stack.Push(root);
-                    yield return root;
-                }
-                else if (stack.Peek().NextSibling() != null)
+                var topElement = stack.Peek();
+                var firstChildInTopElement = topElement.FirstChild;
+
+                if (topElement == root && firstChildInTopElement != null)
                 {
-                    root = stack.Peek().NextSibling();
-                    stack.Pop();
+                    root = firstChildInTopElement;
                     stack.Push(root);
                     yield return root;
                 }
                 else
                 {
-                    stack.Pop();
-                    if (stack.Count == 0)
+                    var nextSiblingElement = topElement.NextSibling();
+                    if (nextSiblingElement != null)
                     {
-                        yield break;
+                        root = nextSiblingElement;
+                        stack.Pop();
+                        stack.Push(root);
+                        yield return root;
+                    }
+                    else
+                    {
+                        stack.Pop();
+                        if (stack.Count == 0)
```

Comment:
Can this be removed at the `while(true)` be replaces with `while(stack.Count>0)`?